app/frontend/private/pos/payment/script.js

```diff
@@ -77,25 +82,57 @@ fetch(`/api/orders/${orderId}`, {
                 })
                     .then((response) => response.json())
                     .then((data) => {
-                        discount = data[0].discount;
-                        discountElement.textContent = `$ ${discount.toFixed(2)}`;
-                        updateCalculations(subtotal, discount);
+                        discountPercentage = data[0].discount;
+                        discountElement.textContent = `$ ${discountPercentage.toFixed(2)}`;
+                        updateCalculations(subtotal, discountPercentage);
                     })
                     .catch((error) => {
                         console.log(error);
                         discountElement.textContent = "$0.00";
-                        updateCalculations(subtotal, discount);
+                        updateCalculations(subtotal, discountPercentage, tipPercentage);
                     });
             } else {
                 discountElement.textContent = "$0.00";
-                updateCalculations(subtotal, discount);
+                updateCalculations(subtotal, discountPercentage, tipPercentage);
             }
         });
-        cashInput.addEventListener("input", () => {
+
+        // tip calculation
+        tipBtns.forEach((button) => {
+            button.addEventListener("click", function () {
+                let tipPercentage = button.getAttribute("data-value");
+                updateCalculations(subtotal, discountPercentage, tipPercentage)
+            });
+        });
+
+        payBtn.addEventListener("click", () => {
             let cash = parseFloat(cashInput.value.replace("$", ""));
-            if (isNaN(cash)) cash = 0;
             const change = cash - total;
-            changeElement.textContent = "$ " + change.toFixed(2);
+            changeElement = "$ " + change.toFixed(2);
+    
+            if (cash < total) {
+                alert("Error: Payment was not fulfilled")
+                return;
+            } else {
+                isPaid = true;
+                const orderDetails = {discountAmount, tipAmount, total, isPaid};
+                console.log(orderDetails);
+                fetch(`/api/orders/process/${orderId}`, {
+                    method: "PUT",
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                    body: JSON.stringify(orderDetails),
+                })
+                    .then((response) => {
+                        if (response.ok) {
```

Comment:
I'd change it to this:

response.ok ? alert(`Your change is $${change.toFixed(2)}`) : alert("Error: Payment could not be processed");